pandas/core/ops/array_ops.py

```diff
@@ -199,7 +207,9 @@ def arithmetic_op(left: ArrayLike, right: Any, op):
     rvalues = ensure_wrapped_if_datetimelike(right)
     rvalues = _maybe_upcast_for_op(rvalues, lvalues.shape)
 
-    if should_extension_dispatch(lvalues, rvalues) or isinstance(rvalues, Timedelta):
+    if should_extension_dispatch(lvalues, rvalues) or isinstance(
+        rvalues, (Timedelta, BaseOffset, Timestamp, NaTType)
```

Comment:
this additional check could maybe be moved into `should_extension_dispatch` (although it is not necessarily related to "is extension array", but rather to "don't take the numexpr path")

Reply:
yah, IIRC should_extension_dispatch is only used here, so might as well refactor/rename/move/whatever is most convenient.

There is a comment below about why Timedelta is included; can you update it for the others

i think check `rvalues is NaT` rather than isinstance check

Reply:
Changed to `right is NaT` and updated the comment.